sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -214,6 +215,7 @@ trait HashJoin {
     val resultProj = createResultProjection
     joinedIter.map { r =>
       numOutputRows += 1
+      avgHashProbe.set(hashed.getAverageProbesPerLookup())
```

Comment:
Ok. I also use TaskCompletionListener to update hash probe at the end of task for  non-whole-stage Join.